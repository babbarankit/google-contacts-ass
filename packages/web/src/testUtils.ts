import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

const Providers = ({ children }) => {
  return children;
  // return (
  //   <ThemeProvider theme="light">
  //     <TranslationProvider messages={defaultStrings}>
  //       {children}
  //     </TranslationProvider>
  //   </ThemeProvider>
  // )
};

const customRender = (ui, options = {}) => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
