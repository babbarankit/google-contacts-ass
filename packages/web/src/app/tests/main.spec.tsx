import React from 'react';
import { render, act } from '../../testUtils';

import Main from '../main';
const Screen = () => (
  <Main>
    <p>Screen Text</p>
  </Main>
);

describe('main render screen', () => {
  it('renders screen', () => {
    const { getByText } = render(<Screen />);
    expect(getByText('Screen Text')).toBeInTheDocument();
  });

  it('health screen macthes snapshot', async () => {
    await act(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
    const { container } = render(<Screen />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
