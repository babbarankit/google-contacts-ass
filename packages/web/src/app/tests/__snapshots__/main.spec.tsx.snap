// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main render screen health screen macthes snapshot 1`] = `
<main>
  <p>
    Screen Text
  </p>
</main>
`;
